tests/integration/sync/test_sync_command.py

```diff
@@ -0,0 +1,398 @@
+
+        # Run infra sync
+        sync_command_list = self.get_sync_command_list(
+            template_file=template_path,
+            code=False,
+            watch=False,
+            resource_id=None,
+            resource=None,
+            dependency_layer=True,
+            stack_name=stack_name,
+            region=None,
+            profile=None,
+            parameter_overrides="Parameter=Clarity",
+            base_dir=None,
+            image_repository=self.ecr_repo_name,
+            image_repositories=None,
+            s3_prefix="integ_deploy",
+            kms_key_id=self.kms_key,
+            capabilities="CAPABILITY_IAM",
+            capabilities_list=None,
+            role_arn=None,
+            notification_arns=None,
+            tags="integ=true clarity=yes foo_bar=baz",
+            metadata=None,
+        )
+
+        sync_process_execute = run_command_with_input(sync_command_list, "y\n".encode())
+        self.assertEqual(sync_process_execute.process.returncode, 1)
+        self.assertIn(
+            "An error occurred (InsufficientCapabilitiesException) when calling the CreateStack operation: \
+Requires capabilities : [CAPABILITY_AUTO_EXPAND]",
+            str(sync_process_execute.stderr),
+        )
+
+    # @parameterized.expand(["template.yaml"])
+    # def test_sync_infra_no_image_repo(self, template_file):
+    #     template_path = str(self.test_data_path.joinpath(template_file))
+    #     stack_name = self._method_to_stack_name(self.id())
+
+    #     # Run infra sync
+    #     sync_command_list = self.get_sync_command_list(
+    #         template_file=template_path,
+    #         code=False,
+    #         watch=False,
+    #         resource_id=None,
+    #         resource=None,
+    #         dependency_layer=True,
+    #         stack_name=stack_name,
+    #         region=None,
+    #         profile=None,
+    #         parameter_overrides="Parameter=Clarity",
+    #         base_dir=None,
+    #         image_repository=None,
+    #         image_repositories=None,
+    #         s3_prefix="integ_deploy",
+    #         kms_key_id=self.kms_key,
+    #         capabilities=None,
+    #         capabilities_list=None,
+    #         role_arn=None,
+    #         notification_arns=None,
+    #         tags="integ=true clarity=yes foo_bar=baz",
+    #         metadata=None,
+    #     )
+
+    #     sync_process_execute = run_command_with_input(sync_command_list, "y\n".encode())
+    #     self.assertEqual(sync_process_execute.process.returncode, 2)
+    #     self.assertIn("", str(sync_process_execute.stderr))
+
+    def _method_to_stack_name(self, method_name):
+        """Method expects method name which can be a full path. Eg: test.integration.test_deploy_command.method_name"""
+        method_name = method_name.split(".")[-1]
+        return f"{method_name.replace('_', '-')}-{CFN_PYTHON_VERSION_SUFFIX}"
+
+    def _stack_name_to_companion_stack(self, stack_name):
```

Comment:
Can we refactor `_stack_name_to_companion_stack` and `_delete_companion_stack` from both deploy and sync into a common file?
Applies to other common methods as well.

Reply:
Good suggestion. Putting it inside package integ base, since they both inherit this class